src/Symfony/Component/Process/Process.php

```diff
@@ -1548,6 +1549,12 @@ private function buildShellCommandline(string|array $commandline): string
 
     private function prepareWindowsCommandLine(string|array $cmd, array &$env): string
     {
+        if (\is_array($cmd) && ($cmd = array_is_list($cmd) ? $cmd : array_values($cmd)) && \strlen($cmd[0]) === strcspn($cmd[0], ':/\\')) {
```

Comment:
Maybe this instead of scanning the whole array?

```suggestion
        if (\is_array($cmd) && ($cmd = isset($cmd[0]) ? $cmd : array_values($cmd)) && \strlen($cmd[0]) === strcspn($cmd[0], ':/\\')) {
```

Reply:
I did it this way because maybe index 0 is not the first...

Reply:
Right I get that but I mean isset() should be faster than array_is_list which has to loop the entire array. And in both cases either 0 is there and you use it, or you call array_values to reset the whole array, that does not change.

Reply:
if the key `0` is set but it is not the first key in the array, using `isset` would use the `0` key not being the first one. On the other hand, the `array_is_list` check will return `false` if you have an array where the key `0` is not the first one (being a list requires using keys going from `0` to `count - 1` **in the right order**)

Reply:
Oh I see, key ordering.. I didn't consider that, makes sense.

Reply:
But I'm removing the array_is_list check because the php engine already implements this fast path in array_values

Reply:
Actually I figured out that we already se array_values on L313 so this check is not needed.

Reply:
If we rely on getting a list to have the proper behavior, I would suggest adding some phpdoc documenting the type as `string|list<string>`. This way, the psalm job would report an error if someone removes the `array_values` usage without being aware that this private method expects that the array is a list.